### PR TITLE
fixed file url issue

### PR DESCRIPTION
**Pull request by tfinnberg**
fix to issue https://github.com/laurent22/joplin/issues/1012. File URLs should work now.

### CODE REVIEW

File: ReactNativeClient/lib/string-utils.js [hunk 1]
```diff
@@ -212,7 +212,7 @@ function toTitleCase(string) {
 }
 
 function urlDecode(string) {
-	return decodeURIComponent((string+'').replace(/\+/g, '%20'));
```
[the remark this answers is not in the document]
Comment (tfinnberg) [reply ReactNativeClient/lib/string-utils.js]: well, I can't think of a reason to replace a '+' character with a space in _string_. Am I missing something?

**Comment by tessus** [comment]
I'm not sure what is happening, but my comments in the review are `pending`. Therefore I don't know, if you have even seen my first comment:

The reason for replacing a + with %20 is so that you can write a local file with + instead of a space and it still works. e.g.: file:///tmp/file+with+space.jpg

I believe some wikis do the same. confluence maybe?

**Comment by tfinnberg** [comment]
@tessus you are absolutely right!  The ' (single quote) is a problem. So far I have not find another critical character, but I will have a closer look soon.
Thank you for explaining the reason for replacing the + with a space. But we will run into a problem if the filename actually contains a +, right?

**Comment by tessus** [comment]
> But we will run into a problem if the filename actually contains a +, right?

Yes  and no. The `+` character is just a short form for `%20`. Theoretically URLs are always supposed to be encoded, making `+` is just an exception.

If a file really had a `+` in its name, you'd have to encode it with `%2B`.

**Comment by tfinnberg** [comment]
@tessus yes, moving the discussion to a new issue is a good idea. I have changed the PR to only cover the basic module export part, leaving the + to %20 swap untouched. Thanks!

**Comment by tessus** [comment]
@tfinnberg I've opened a new issue #1074 

Unfortunately the code has changed since you submitted the PR and there's now a conflict. Please rebase and squash, or create a new PR, if that's easier.

**Comment by tfinnberg** [comment]
@tessus ok, I have fixed the PR

**Comment by tessus** [comment]
@laurent22 This PR is ready to be merged. Any objections, if I do so?

**Comment by laurent22** [comment]
Looks good, thank you!

**Pull request merged**